test/extensions/filters/http/wasm/wasm_filter_stress_test.cc

```diff
@@ -0,0 +1,234 @@
+                                       wasm_vm, wasm_file));
+
+  //
+  // This test first sends a request from wasm to a callout cluster:
+  //    Client -> Envoy -> Wasm -> Callout Cluster
+  // Envoy then unconditionally forwards the request to the origin cluster
+  //              Envoy -> Origin Cluster
+  //
+  // Both Callout and Origin clusters have a single server that immediately
+  // returns a 200 OK response.
+  //
+  LoadGeneratorPtr client = startServers(
+      {new ClusterHelper(StressTest::ORIGIN_CLUSTER_NAME,
+                         {new ServerCallbackHelper([](ServerConnection&, ServerStream& stream,
+                                                      Envoy::Http::HeaderMapPtr&&) {
+                           ENVOY_LOG(debug, "Origin server received request");
+                           Http::TestHeaderMapImpl response{{":status", "200"}};
+                           stream.sendResponseHeaders(response);
+                         })}),
+       new ClusterHelper(callout_cluster_name,
+                         {new ServerCallbackHelper([](ServerConnection&, ServerStream& stream,
+                                                      Envoy::Http::HeaderMapPtr&&) {
+                           ENVOY_LOG(debug, "Callout server received request");
+                           Envoy::ProtobufWkt::Value response;
+                           response.set_string_value("response");
+                           stream.sendGrpcResponse(Envoy::Grpc::Status::Ok, response);
+                         })})});
+
+  //
+  // Exec test and wait for it to finish
+  //
+
+  Envoy::Http::HeaderMapPtr request{new Envoy::Http::TestHeaderMapImpl{
+      {":method", "GET"}, {":path", "/"}, {":scheme", "http"}, {":authority", "host"}}};
+  client->run(connections_to_initiate, requests_to_send, std::move(request));
+
+  //
+  // Block until all servers exit
+  //
+
+  stopServers();
+
+  //
+  // Evaluate test
+  //
+
+  // All client connections are successfully established.
+  EXPECT_EQ(client->connectSuccesses(), connections_to_initiate);
+  EXPECT_EQ(client->connectFailures(), 0);
+  // Client close callback called for every client connection.
+  EXPECT_EQ(client->localCloses(), connections_to_initiate);
+  // Client response callback is called for every request sent
+  EXPECT_EQ(client->responsesReceived(), requests_to_send);
+  // Every response was a 2xx class
+  EXPECT_EQ(client->class2xxResponses(), requests_to_send);
+  EXPECT_EQ(client->class4xxResponses(), 0);
+  EXPECT_EQ(client->class5xxResponses(), 0);
+  EXPECT_EQ(client->responseTimeouts(), 0);
+  // No client sockets are rudely closed by server / no client sockets are
+  // reset.
+  EXPECT_EQ(client->remoteCloses(), 0);
+
+  // assert that the callout server and origin server see every request
+  EXPECT_EQ(clusterHelper(StressTest::ORIGIN_CLUSTER_NAME).requestsReceived(), requests_to_send);
+  EXPECT_EQ(clusterHelper(callout_cluster_name).requestsReceived(), requests_to_send);
```

Comment:
Assertions here can be built out once the gRPC callout response is handled correctly.  Counters can also be dumped and asserted on (see StressTest superclass)